app/navigators/ChargeDNavigator.scala

```diff
@@ -114,14 +115,32 @@ class ChargeDNavigator @Inject()(val dataCacheConnector: UserAnswersCacheConnect
     case TaxQuarterReportedAndPaidPage(ChargeTypeLifetimeAllowance, index, schemeIndex) =>
       controllers.mccloud.routes.ChargeAmountReportedController
         .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index, schemeIndex)
-    case ChargeAmountReportedPage(ChargeTypeLifetimeAllowance, index, _) =>
-      CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+    case ChargeAmountReportedPage(ChargeTypeLifetimeAllowance, index, schemeIndex) =>
+      val schemeSize = countSchemeSize(ua, index)
+      val schemeSizeLessThan5 = schemeSize > 0 && schemeSize < 5
+      (schemeIndex, schemeSizeLessThan5) match {
+        case (Some(i), true) => controllers.mccloud.routes.AddAnotherPensionSchemeController
+          .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index, i)
+        case (_, false)  => CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+      }
+
+    case AddAnotherPensionSchemePage(ChargeTypeLifetimeAllowance, index, schemeIndex) =>
+      ua.get(AddAnotherPensionSchemePage(ChargeTypeLifetimeAllowance, index, schemeIndex)) match {
+        case Some(true) => controllers.mccloud.routes.EnterPstrController
+          .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index, countSchemeSize(ua, index))
+        case Some(false) => CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+      }
 
     case CheckYourAnswersPage => AddMembersController.onPageLoad(srn, startDate, accessType, version)
     case AddMembersPage => addMembers(ua, srn, startDate, accessType, version)
     case DeleteMemberPage => deleteMemberRoutes(ua, srn, startDate, accessType, version)
   }
 
+  private def countSchemeSize(userAnswers: UserAnswers, index :Int): Int = {
```

Comment:
I feel it's better to try to keep any JsPaths inside Page objects. If need be you could create a Query object for this, see for example LifetimeAllowanceMembersQuery

Reply:
Moved JsPath to SchemesQuery